online-shop/src/app/modules/shared/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts

```diff
@@ -1,19 +1,61 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, Output, EventEmitter } from '@angular/core';
 import { Product } from 'src/app/modules/shared/types/products.types';
-import { ShoppingCartService } from 'src/app/services/shopping-cart-service';
+import { ShoppingCartService } from 'src/app/modules/shared/shopping-cart/services/shopping-cart-service';
+import { OrderCreateDto } from 'src/app/modules/shared/types/order-create-dto.types';
+import { ProductIdAndQuantity } from 'src/app/modules/shared/types/product-id-and-quantity.types';
+
 @Component({
   selector: 'app-shopping-cart-details-view',
   templateUrl: './shopping-cart-details-view.component.html',
   styleUrls: ['./shopping-cart-details-view.component.scss'],
 })
 export class ShoppingCartDetailsViewComponent {
   @Input() products!: Product[];
+  @Output() placeOrder = new EventEmitter<OrderCreateDto>();
+  productIdAndQuantities: ProductIdAndQuantity[] = [];
 
-  constructor(private cartService: ShoppingCartService) {} // Inject the ShoppingCartService
+  constructor(private cartService: ShoppingCartService) {}
 
   onRemoveButtonClick(product: Product) {
     console.log('Removed');
     this.cartService.removeFromCart(product.id);
-    this.products = this.cartService.getCartItems(); // Update the products list after removing the item
+    this.products = this.cartService.getCartItems();
+  }
+
+  onUpdateQuantity(productId: string, quantity: number) {
+    // Check if the productId already exists in the list
+    if (quantity > 0) {
```

Comment:
Nothing wrong, just a personal preference.
You could check the negated statement and directly return from the method. Something like 
```typescript if (quantity <= 0) return; ```
and then continue with the logic you implemented for quantity > 0. In this way the code is less imbricated.

Reply:
Now doing the way you suggested.